tests/model_registry/test_user_permission.py

```diff
@@ -0,0 +1,292 @@
+            admin_client=admin_client,
+            context=context_manager,
+            mr_instance=model_registry_instance,
+            mr_namespace_name=model_registry_namespace,
+        )
+
+    @pytest.mark.smoke
+    @pytest.mark.parametrize(
+        "user_name, password",
+        [
+            ("ldap-user1", os.environ.get("NON_ADMIN_PASSWORD")),
+        ],
+    )
+    def test_user_added_to_group(
+        self: Self,
+        updated_dsc_component_state_scope_class: Namespace,
+        model_registry_instance: ModelRegistry,
+        model_registry_namespace: str,
+        admin_client: DynamicClient,
+        user_name: str,
+        password: str,
+        user_in_group_context: Callable[[str], ContextManager],
+    ):
+        """
+        Test that a user cannot access the Model Registry before being added to a group,
+        and can access it after being added to the group.
+        """
+        assert model_registry_instance.name == MR_INSTANCE_NAME
+
+        user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
+
+        LOGGER.info("User has no access to MR")
+        assert_mr_client(
+            user_token=user_token,
+            admin_client=admin_client,
+            context=pytest.raises(ForbiddenException),
+            mr_instance=model_registry_instance,
+            mr_namespace_name=model_registry_namespace,
+        )
+
+        LOGGER.info("Add user to the model registry users group")
+        with user_in_group_context(user_name):
+            user_token = get_token(user_name=user_name, password=password, admin_client=admin_client)
+
+            LOGGER.info("User has access to MR")
+            assert_mr_client(
+                user_token=user_token,
+                admin_client=admin_client,
+                context=nullcontext(),
+                mr_instance=model_registry_instance,
+                mr_namespace_name=model_registry_namespace,
+            )
+
+    @pytest.mark.smoke
+    @pytest.mark.parametrize(
+        "user_name, password, new_group",
+        [
+            ("ldap-user1", os.environ.get("NON_ADMIN_PASSWORD"), (NEW_GROUP_NAME, "ldap-user1")),
+        ],
+        indirect=["new_group"],
+    )
+    def test_create_group(
+        self: Self,
+        updated_dsc_component_state_scope_class: Namespace,
+        model_registry_instance: ModelRegistry,
+        model_registry_namespace: str,
+        admin_client: DynamicClient,
+        user_name: str,
+        password: str,
+        new_group: str,
+    ):
+        """
+        Test creating a group, granting it model registry access, and verifying user access.
+        """
+
+        LOGGER.info("Group created and user added to it")
+
+        with RoleBinding(
+            client=admin_client,
+            namespace=model_registry_namespace,
+            name="test-model-registry-group-edit",
+            role_ref_name="edit",
+            role_ref_kind="ClusterRole",
```

Comment:
Where do you create this role?

Reply:
is it not created by default? `oc get clusterroles -o yaml | grep -A 10 modelregistry`

Reply:
The `edit` role present by default on the cluster is not the Role we want to use, since it has a ton of permissions for all kinds of different resources. The Role we want here should only give permission to get the MR resources we need, i.e.
```
rules:
 - verbs:
 	- get
   apiGroups:
 	- ''
   resources:
 	- services
   resourceNames:
 	- <MR_NAME>
 - verbs:
 	- get
   apiGroups:
 	- ''
   resources:
 	- endpoints
   resourceNames:
 	- <MR_NAME>
```